### ISSUE TITLE
What resources do we need for on-boarding new front-end developers?

### ISSUE THREAD

**Issue by mcnasby** [backupify/styleguide#12]
Also, what resources are needed for properly involving full-stack developers in the UI development process to some degree?

Consider the scenario where we on-board 4 new front-end developers in month. What are the most critical resources needed to make sure they start down the right path -- toward proper coordination?

**Comment by jbarreiros** [comment]
Hm, it depends on the project, right?

The obvious are:
* The shared vision by all front-end devs
* Documentation like common tools, example configs, style conventions
* Each project must have a well-written and **up-to-date** README explaining, at a minimum, how to set up the project

We should probably take stock of what tools we are using in our projects. That could help determine what kind of resources may be needed.

Really, the key to helping someone understand a project is proper documentation. Reality though is that maintaining up-to-date documentation is really really hard.

**Comment by hoodsy** [comment]
Totally agree with documentation being the most important for getting people up and running in projects, new hire or not. I know @diegoconcha and @kswin have begun standardizing around ngDocs, which is built on **JSDoc**. John, I remember you and @bkotos mentioning JSDoc and seeing it in your code a bit.

It would be great to potentially standardize around JSDoc. Using it is pretty self-explanatory, but we could figure out a basic system / minimum requirements for describing code. For example, requiring @description, @params, @return tags for each function (or something to this end).

In terms of style conventions, I know that Kim and Diego use JSHint and JSCS. Sharing the config files for these linters across projects could go a long way towards front end code looking like it **comes from one entity.**

@jbarreiros @mcnasby

**Comment by jbarreiros** [comment]
@hoodsy Yes, we use JSDoc extensively and I try to strike the right balance. 

At a minimum, I expect 
* Every method to have its input (`@param`) and outputs (`@return`) documented. If the method name is self-explanatory, than I don't think it necessarily needs a description.
* All object properties to be documented (`@name`, `@type`).
* While JS doesn't have a concept of *private*, I do hint at methods and properties being "private" with `@private`. We also prepend the property or method name with an underscore.

Another jsdoc tag I use a lot is `@listens` and `@fires`. Backbonejs is very event-based and those tags provide an easy way to search where an event originates and what listens to it.

It's easy for documentation to get out of date, so the goal is to write code that is easy to follow/understand and only document the *weird shit*. I have no problem with long variable names for example. Additionally, I try not to use docblocks to explain concepts. Instead, I have a section in [HomePool](http://homepool.datto.lan/display/DNA/Cloud-UI+Guides) where I elaborate on larger concepts. It is a little sparse right now.
